### ISSUE TITLE
eglGetProcAddress returns NULL for GLES functions, while those functions are actually callable

### ISSUE THREAD

**Issue by kaaass** [GrapheneCt/PVR_PSP2#18]
Hi, I'm trying to port a game using PVR_PSP2, which loads GL functions dynamically but found eglGetProcAddress didn't return right address. The situation is:

```cpp
init_pvr_psp2();
init_egl();
auto addr = reinterpret_cast<uint32_t>(eglGetProcAddress("glGetError"));
log("glGetError addr = %x, error = %x\n", addr, eglGetError());
// prints: glGetError addr = 0, error = 3000
log("glGetError = %d\n", glGetError());
// prints: glGetError = 0, might indicate that GLES module are loaded
addr = reinterpret_cast<uint32_t>(eglGetProcAddress("glGetError"));
log("glGetError addr = %x, error = %x\n", addr, eglGetError());
// prints: glGetError addr = 0, error = 3000
```

The complete test code can be found [here](https://gist.github.com/kaaass/b732443f4be7e67c40542e470b1dcb08). Additionally, loading libGLESv1_CM.suprx manually in `init_pvr_psp2` would solve this problem. But I thought this might not be the correct way to use it. Anyway, thanks for developing this excellent repo! Any help will be appreciated.

**Comment by isage** [comment]
You are using gles2 context. Why are you trying to load gles1 module?

**Comment by isage** [comment]
Moreover, if you intend to use sdl2, like your code suggests - why not use sdl context creation funcs?

**Comment by kaaass** [comment]
> You are using gles2 context. Why are you trying to load gles1 module?

I had tried to load libGLESv2.suprx manually, but it crashed. Also, yes, loading libGLESv1_CM.suprx did make eglGetProcAddress return a non-null value, but those functions didn't seem to work correctly.

> Moreover, if you intend to use sdl2, like your code suggests - why not use sdl context creation funcs?

Actually, this is how I found this issue. Take my SDL2 + gl4es4vita code as an example. When I comment [this line](https://github.com/kaaass/gl4es4vita-sdl2-gear/blob/93fe3813734f5c353a79a3ca3a2c30edc7af3afa/main.cpp#L310) to let SDL2 handles context creation stuff, the program crashed during OpenGL calls. I tried to recompile gl4es4vita in debug profile and found it complains that `glGetError` is NULL [here](https://github.com/kaaass/gl4es4vita/blob/811b9fdec35299af54e6f45156aad6de082c3d99/src/glx/hardext.c#L248) during `SDL_GL_CreateContext`. So I extracted several vital codes and managed to reproduce this.

I guess I did sth wrong while setting up the lib (SDL2 + gl4es4vita + PVR_PSP2 combo). But for the use case metioned in this issue, NULL return of `eglGetProcAddress` makes no sense to me (or maybe I miss sth in initialization?).
